src/System.Memory/src/System/SpanHelpers.BinarySearch.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class SpanHelpers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int BinarySearch<T, TComparable>(
+            this ReadOnlySpan<T> span, TComparable comparable)
+            where TComparable : IComparable<T>
+        {
+            if (comparable == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.comparable);
+            // TODO: Make `ref readonly`/`in` when Unsafe.Add(ReadOnly) supports it
+            return BinarySearch(ref span.DangerousGetPinnableReference(), span.Length, comparable);
+        }
+
+        // TODO: Make s `ref readonly`/`in` when Unsafe.Add(ReadOnly) supports it
+        internal static int BinarySearch<T, TComparable>(
+            ref T s, int length, TComparable comparable) 
```

Comment:
nit: can u use a more descriptive arg name (and local vars name) here instead of single character names (like 's') please?